overlay.yml

```diff
@@ -0,0 +1,122 @@
+overlay: 1.0.0
+info:
+  title: "passage-go-specific codegen updates via the OpenAPI Overlay specification (https://github.com/OAI/Overlay-Specification)"
+  version: 1.0.0
+actions:
+
+# rename types
+
+- target: $.components.schemas.CreateUserRequest
+  description: Rename the generated Go type name
+  update:
+    x-go-name: CreateUserArgs
+
+- target: $.components.schemas.UpdateUserRequest
+  description: Rename the generated Go type name
+  update:
+    x-go-name: UpdateUserOptions
+
+- target: $.components.schemas.ListPaginatedUsersResponse
+  description: Rename the generated Go type name
+  update:
+    x-go-name: PaginatedUsersResponse
+
+- target: $.components.schemas.UserInfo
+  description: Rename the generated Go type name
+  update:
+    x-go-name: PassageUser
+
+- target: $.components.schemas.MagicLinkChannel
+  description: Rename the generated Go type name
+  update:
+    x-go-name: ChannelType
+
+- target: $.components.schemas.CreateMagicLinkRequest
+  description: Rename the generated Go type name and make it private
+  update:
+    x-go-name: magicLinkArgs
+
+- target: $.components.schemas.ListPaginatedUsersResponse
+  description: Rename the generated Go type name and make it private
+  update:
+    x-go-name: paginatedUsersResponse
+
+# rename enums
```

Comment:
https://github.com/oapi-codegen/oapi-codegen/tree/v2.4.1#openapi-extensions:~:text=x%2Denum%2Dvarnames%20/%20x%2DenumNames

chose to use `x-enumNames` since it looks like that's the preferred extension name
- [source](https://github.com/oapi-codegen/oapi-codegen/blob/9feb3033b6127185b7d89e240bcf5471bf8328f4/pkg/codegen/schema.go#L490-L498)
- [example](https://github.com/oapi-codegen/oapi-codegen/tree/v2.4.1/examples/extensions) (no example for varnames)